### ISSUE TITLE
[PLUGIN, TEMPORAL] After the activity is completed and worker is terminated (graceful), the last event is still ActivityTaskScheduled

### ISSUE THREAD

**Issue by khoerintus** [spiral/roadrunner-plugins#12]
Hi @rustatian , thanks for the fix.

But I encountered another issue.
When the activity is complete and the worker is terminated, but in temporal web, the last event in the workflow is still ActivityTaskScheduled.
I expected when the activity is complete and worker is successfully terminated, I can see the last event in temporal web is ActivityTaskCompleted.

I already update my samples code [here](https://github.com/khoerintus/samples-php), if you need it.
I just lower the sleep to 20s.

_Originally posted by @khoerintus in https://github.com/spiral/roadrunner/issues/610#issuecomment-827694080_

**Comment by changwuf31** [comment]
Hi @rustatian, FYI, https://github.com/temporalio/sdk-go/releases/tag/v1.11.0 has been released.

**Comment by rustatian** [comment]
Thanks, I know, updating the rr-temporal: https://github.com/temporalio/roadrunner-temporal/pull/110

**Comment by changwuf31** [comment]
Hi @rustatian, anything we can do to help with the PR ?

**Comment by rustatian** [comment]
I'm recovering from the COVID-19, so, 2 weeks at least.

**Comment by changwuf31** [comment]
Hope you'll have a speedy recovery 🙏

**Comment by rustatian** [comment]
@changwuf31 We are blocked by the https://github.com/temporalio/sdk-go/issues/636 and this change https://github.com/temporalio/sdk-go/commit/16bb227d85dea95ed89b1fee6d7e04438b50103d. We can't update to v 1.11 and upper with this change, so, let's wait until this issue will be resolved.

**Comment by rustatian** [comment]
@khoerintus , @changwuf31 Updated, could you try the https://github.com/spiral/roadrunner-binary/releases/tag/v2.6.0-rc.1 release? This bug should be fixed in it. temporal go-sdk updated to the latest version.

**Comment by changwuf31** [comment]
Hi @rustatian 
I've tried the 2.6.0-rc1

The issue is still unresolved yet though ..
Got this in the log
```
2021-11-29T15:42:30.452Z	INFO	temporal    	Task processing failed with error	{"Namespace": "default", "TaskQueue": "default", "WorkerID": "1197313@fedora@default@2", "WorkerType": "ActivityWorker", "Error": "grpc: the client connection is closing"}
```

**Comment by rustatian** [comment]
@changwuf31 It's the correct message. What's the last status of the activity?

**Comment by changwuf31** [comment]
> @changwuf31 It's the correct message. What's the last status of the activity?

Log
![image](https://user-images.githubusercontent.com/2149047/143899460-36845779-edf2-42d3-995b-edeec0a9ca56.png)
Activity
![image](https://user-images.githubusercontent.com/2149047/143899604-1a32a2dc-6395-4786-99ab-c18d2d973dbd.png)
Workflow 
![image](https://user-images.githubusercontent.com/2149047/143899681-523abef0-7c7c-4a7d-9b5f-30ab98d65170.png)

**Comment by changwuf31** [comment]
```
# Application configuration
rpc:
    listen: tcp://127.0.0.1:6001
    # listen: ${ROADRUNNER_RPC_URL}

server:
    command: "php artisan temporal-worker:simple"
    env:
        DEBUG: true

# Workflow and activity mesh service
temporal:
    address: "localhost:7233"
    # address: ${TEMPORAL_GRPC_URL}
    activities:
        num_workers: 3
        # default 0 - no limit
        max_jobs: 50
        # default 1 minute
        allocate_timeout: 60s
        # default 1 minute
        destroy_timeout: 60s
        # supervisor used to control http workers
        supervisor:
            # watch_tick defines how often to check the state of the workers (seconds)
            watch_tick: 1s
            # ttl defines maximum time worker is allowed to live (seconds)
            ttl: 0
            # idle_ttl defines maximum duration worker can spend in idle mode after first use. Disabled when 0 (seconds)
            idle_ttl: 0
            # exec_ttl defines maximum lifetime per job (seconds)
            exec_ttl: 0
            # max_worker_memory limits memory usage per worker (MB)
            max_worker_memory: 128


logs:
    # development / production
    mode: development
    level: debug
    encoding: console
    #channels:
       #activities:
          #mode: none
       #workflows:
          #mode: none

endure:
  # default 30s to finish, possible values 10s,100m,5h
  grace_period: 120s
  # default false, print graph in the graphviz format to the stdout (paste here to visualize https://dreampuf.github.io)
  print_graph: false
  # default error
  log_level: debug
  ````
  Log with debug level
  
![image](https://user-images.githubusercontent.com/2149047/143901440-b38cea82-4243-4f2b-8b98-8e67ab53f51f.png)

**Comment by rustatian** [comment]
Ok, thanks, I'll have a look. Looks like the `client` plugin stops first, preventing other plugins to send the data to the temporal.

**Comment by rustatian** [comment]
@changwuf31 Could you try this version https://github.com/spiral/roadrunner-binary/releases/tag/v2.6.0, I fixed the plugin's stop order, now, the client stopped last in the row.

**Comment by changwuf31** [comment]
![image](https://user-images.githubusercontent.com/2149047/144272841-de4e50cc-cbc8-42eb-a024-ce4e50bb20c5.png)

The last activity is "ActivityTaskCompleted"
The workflow status is "COMPLETED"
The client plugin also is stopped after the workflow plugin.
Looks Good, thank u @rustatian  :+1:

**Comment by rustatian** [comment]
Awesome, thank you @changwuf31 . Closing the ticket.

**Comment by rustatian** [comment]
link: https://github.com/spiral/endure/pull/109
